src/protocol/messages/delete_records.rs

```diff
@@ -0,0 +1,294 @@
+use std::io::{Read, Write};
+
+use crate::protocol::{
+    api_key::ApiKey,
+    api_version::{ApiVersion, ApiVersionRange},
+    error::Error,
+    messages::{
+        read_compact_versioned_array, read_versioned_array, write_compact_versioned_array,
+        write_versioned_array,
+    },
+    primitives::{CompactString, CompactStringRef, Int16, Int32, Int64, String_, TaggedFields},
+    traits::{ReadType, WriteType},
+};
+
+use super::{
+    ReadVersionedError, ReadVersionedType, RequestBody, WriteVersionedError, WriteVersionedType,
+};
+
+#[derive(Debug)]
+pub struct DeletePartitionRequest {
+    /// The partition index.
+    pub partition_index: Int32,
+
+    /// The deletion offset.
+    pub offset: Int64,
+
+    /// The tagged fields.
+    ///
+    /// Added in version 2
+    pub tagged_fields: Option<TaggedFields>,
+}
+
+impl<W> WriteVersionedType<W> for DeletePartitionRequest
+where
+    W: Write,
+{
+    fn write_versioned(
+        &self,
+        writer: &mut W,
+        version: ApiVersion,
+    ) -> Result<(), WriteVersionedError> {
+        let v = version.0 .0;
+        assert!(v <= 2);
+
+        self.partition_index.write(writer)?;
+        self.offset.write(writer)?;
+
+        if v >= 2 {
+            match self.tagged_fields.as_ref() {
+                Some(tagged_fields) => {
+                    tagged_fields.write(writer)?;
+                }
+                None => {
+                    TaggedFields::default().write(writer)?;
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
+pub struct DeleteTopicRequest {
```

Comment:
Perhaps `DeleteRecordsRequestTopic` would avoid confusion with the `DeleteTopics` API